pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -542,6 +545,32 @@ func BootstrapKuryr(conf *operv1.NetworkSpec, kubeClient client.Client) (*bootst
 		}
 	}
 
+	octaviaVersion, err := getSavedAnnotation(kubeClient, names.KuryrOctaviaVersionAnnotation)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, errors.Wrap(err, "failed to get kuryr-config ConfigMap")
+	}
+
+	maxOctaviaVersion, err := getMaxOctaviaAPIVersion(lbClient)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get max octavia api version")
+	}
+
+	if octaviaVersion != "" {
```

Comment:
Can you add a comment explaining why we need all this and the logic behind it? For sure it'll be useful in 6 months. ;)

Reply:
Yup.